### ISSUE TITLE
Works only with raspbian

### ISSUE THREAD

**Issue by vallettea** [xdissent/pihotenuse#3]
Have you tried, the lib with other images than raspian ?
I've used [hypriot](http://blog.hypriot.com/post/hypriotos-back-again-with-docker-on-arm/) image for init with:

```
piho init hypriot.img
piho create rpi-docker
```

but:

```
$ piho shell rpi-docker                                                                                      [ruby-2.1.3p242]
Launching shell for rpi-docker
mount: block device /dev/nbd0 is write-protected, mounting read-only
mount: you must specify the filesystem type
```

Where does the limitation to raspian comes from ?

**Comment by vallettea** [comment]
here is the verbose version:

```
piho shell rpi-docker                                                                                      [ruby-2.1.3p242]
Checking for image rpi-docker
Checking for boot2docker
Exporting b2d env vars
Creating boot2docker profile
Checking boot2docker status
Setting docker shell environment
Checking for pihotenuse docker image
Creating debs folder
Creating apt repo config
Checking for pihotenuse disk image
Launching shell for rpi-docker
mount: block device /dev/nbd0 is write-protected, mounting read-only
mount: you must specify the filesystem type
```

**Comment by vallettea** [comment]
I'm trying to document my research. 

The problem seems to be around qemu-nbd. I entered the ubuntu image with : 

```
piho docker run -it -v ~/.pihotenuse/:/piho/home  xdissent/pihotenuse bash
```

and then i try qemu-nbd:

```
root@51007bc2a37c:/piho# qemu-nbd -c /dev/nbd0 build.qcow2
Failed to open /dev/nbd0: No such file or directory/build/buildd/qemu-2.0.0+dfsg/nbd.c:nbd_receive_request():L638: read failed
```

from [this entry](http://blog.loftninjas.org/2008/10/27/mounting-kvm-qcow2-qemu-disk-images/) i supposed I had to do:

```
root@51007bc2a37c:/piho# modprobe nbd max_part=8
modprobe: ERROR: ../libkmod/libkmod.c:556 kmod_search_moddep() could not open moddep file '/lib/modules/4.0.2-boot2docker/modules.dep.bin'
```

why does my ubuntu image knows about boot2docker ?

**Comment by xdissent** [comment]
No not yet, my internet was down all day

**Comment by xdissent** [comment]
Oh no - does the new boot2docker image not ship with the nbd module? The piho setup should take care of a one-time loading of it...  I'll check this out this afternoon.

**Comment by vallettea** [comment]
Hello,  how can i check that `does the new boot2docker image not ship with the nbd module`?

**Comment by xdissent** [comment]
Oh sorry, I misunderstood where the nbd error was coming from. That's an easy fix - change line 666 (heh) and 667 from:

```
mount -o offset=$((512*122880)) "$NBD_DEV" mount
mount -o offset=$((512*8192)) "$NBD_DEV" mount/boot
```

to:

```
mount -o offset=$((512*133120)) "$NBD_DEV" mount
mount -o offset=$((512*2048)) "$NBD_DEV" mount/boot
```

That should correctly mount the img in the docker host. However, the hypriot img doesn't seem to have a default `/etc/ld.so.preload`, which piho currently requires. A quick fix is to add `touch mount/etc/ld.so.preload` before line 674. With those 2 changes in place I was able to boot into hypriot successfully. It looks like there's some bash profile weirdness going on once booted, but it's harmless and should be easy to fix.

The nbd issue will take a little work to gracefully handle other offsets, but checking for `/etc/ld.so.preload` is obviously easy. I've got a meeting this morning but I'll give the offset detection a shot later today so I can push a real patch.

**Comment by xdissent** [comment]
Oh! Remember to run `./piho.sh update` (or if you're modifying the piho in your PATH, `piho update`). Otherwise the docker container will still have the unmodified script and you'll see no change!

**Comment by vallettea** [comment]
testing this right now on another image (odroid for arm7 with ubuntu so it has a /etc/ls.so.preload i guess).
Can you explain how you figured out those numbers ? and if I should change them for the odroid image ?

**Comment by vallettea** [comment]
ps: this patch would be awesome !

**Comment by xdissent** [comment]
Sure, all I did was add `fdisk -l "$NBD_DEV"` before the actual mounting on line 664, then:

```
export PIHO_DEBUG=1
export PIHO_TRACE=1
piho update
piho create testing
piho sh testing
```

The fdisk output should look like:

```
     Device Boot      Start         End      Blocks   Id  System
/dev/nbd0p1            8192      122879       57344    c  W95 FAT32 (LBA)
/dev/nbd0p2          122880     6399999     3138560   83  Linux
```

You're looking for the two numbers in the `Start` column.

**Comment by xdissent** [comment]
Fixed in v0.0.3

**Comment by vallettea** [comment]
:+1:  thx so much!
